pkg/stanza/operator/helper/emitter.go

```diff
@@ -79,15 +78,21 @@ func NewLogEmitter(set component.TelemetrySettings, consumerFunc func(context.Co
 
 // Start starts the goroutine(s) required for this operator
 func (e *LogEmitter) Start(_ operator.Persister) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	e.cancel = cancel
+
 	e.wg.Add(1)
-	go e.flusher()
+	go e.flusher(ctx)
 	return nil
 }
 
 // Stop will close the log channel and stop running goroutines
 func (e *LogEmitter) Stop() error {
 	e.stopOnce.Do(func() {
-		close(e.closeChan)
+		// the cancel func could be nil if the emitter is never started.
```

Comment:
I realized that quite a few unit tests call `Stop()` without actually starting the emitter, and that would fail because the context is not created if `Start()` is not called. I put a defensive check here. 

I can also move the context creation to `NewLogEmitter`, which feels a bit less "natural". Open to suggestions.